submittal/src/main/java/com/kyriba/submittal/api/SpeakerController.java

```diff
@@ -0,0 +1,104 @@
+package com.kyriba.submittal.api;
+
+import com.kyriba.submittal.domain.dto.PersonalInfoDto;
+import com.kyriba.submittal.domain.dto.SpeakerDataDto;
+import com.kyriba.submittal.domain.dto.TopicDto;
+import com.kyriba.submittal.service.SpeakerService;
+import lombok.Builder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.SessionAttribute;
+
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE;
+
+
+/**
+ * @author M-ABL
+ */
+// TODO: setup security. Let's assume a filter retrieves a speaker's id by email/password and puts the id to the session.
+@RestController
+@RequestMapping(value = "/api/v1/speaker",

```

Comment:
Common prefix could be moved to common parent class.

Reply:
Should we specify a name of a service in the url? e.g. /api/v1/submittal and then specific path? Or submittal is redundant in this case?

Reply:
In REST style API we don't specify service name in URL. Instead we operate only with resource names.

Reply:
what's a common part in this case? '/api/v1' ? is it OK when the first controller has 'v1' version and the second 'v2'? If so, then a version is not a common part. Seems only consumes and produces can be moved. But a parent class looks empty then.

Reply:
Usually whole microservice has the same version of API.

Reply:
please clarify one more time: the common part is '/api/v1' and for each method: '/speakers/...' ?

Reply:
Common part for all controllers is /api/v1, common part for SpeakerController is /speakers.

Reply:
noob question then: how to extend RequestMapping annotation in order to combine a path ('api/v1' + '/speakers')? As I know, the path is overridden but not merged

Reply:
One common approach is to use config param for API prefix at class level like @RequestMapping(path = "${api.version.path}/users"). There are some other options: https://stackoverflow.com/questions/28006501/how-to-specify-prefix-for-all-controllers-in-spring-boot